pygit2/remote.py

```diff
@@ -123,6 +123,7 @@ def fetch(self, refspecs=None, message=None, callbacks=None, prune=C.GIT_FETCH_P
         with git_fetch_options(callbacks) as payload:
             opts = payload.fetch_options
             opts.prune = prune
+            opts.proxy_opts.type = C.GIT_PROXY_AUTO
```

Comment:
This should be a function argument, and the default behavior should not be modified.

Looking at the proxy options in libgit2 I think we could do an interface like this:

```
fetch(..., proxy=None) # or False (default)
fetch(..., proxy=True) # for auto
fetch(..., proxy='http://....')
```

Reply:
@jdavid Done (with docs). Please review again.